spec/Cabin/InteriorLights.vspec

```diff
@@ -10,57 +11,56 @@
 # All interior lights and sensors
 #
 
+#
+# Non-configurable lights
+#
 IsGloveBoxOn:
   datatype: boolean
   type: actuator
   description: Is glove box light on
 
-IsTrunkOn:
-  datatype: boolean
-  type: actuator
-  description: Is trunk light light on
-
 IsDomeOn:
   datatype: boolean
   type: actuator
   description: Is central dome light light on
 
-AmbientLight:
+PerceivedAmbientLight:
   datatype: uint8
   type: sensor
   unit: percent
   min: 0
   max: 100
-  description: How much ambient light is detected in cabin. 0 = No ambient light. 100 = Full brightness
-
-LightIntensity:
-  datatype: uint8
-  type: sensor
-  unit: percent
-  min: 0
-  max: 100
-  description: Intensity of the interior lights. 0 = Off. 100 = Full brightness.
+  description: The percentage of ambient light that is measured (e.g., by a sensor) inside the cabin. 0 = No ambient light. 100 = Full brightness.
+  comment: V4.0 named changed from "AmbientLight" to "PerceivedAmbientLight". This is a read-only property that refers to the pre-existing light (e.g., natural light). If you are looking for the in-cabin decorative lights that sometimes are also called "AmbientLights", please refer to the branch Vehicle.Cabin.Light.AmbientLight.
 
 #
-# First / front row lights
+# Configurable lights
 #
 Spotlight:
   type: branch
-  instances: Row[1,4]
+  instances: 
+    - Row[1,2]
+    - ["Driver","Passenger"]
   description: Spotlight for a specific area in the vehicle.
+#include SingleConfigurableLight.vspec Spotlight
 
-Spotlight.IsSharedOn:
-  datatype: boolean
-  type: sensor
-  description: Is a shared light across a specific row on
 
-Spotlight.IsLeftOn:
-  datatype: boolean
-  type: actuator
-  description: Is light on the left side switched on
+AmbientLight:
+  type: branch
+  instances: 
+    - Row[1,2]
+    - ["Driver","Passenger"]
+  description: Decorative coloured light inside the cabin, usually mounted on the door, cealing, etc.
+#include SingleConfigurableLight.vspec AmbientLight
 
+InteractiveLightBar:
+  type: branch
+  description: Decorative coloured light bar that supports effects. It is usually mounted on the dashboard (e.g., BMW i7 Interactive bar).
+#include SingleConfigurableLight.vspec InteractiveLightBar
 
-Spotlight.IsRightOn:
-  datatype: boolean
+InteractiveLightBar.Effect:
   type: actuator
-  description: Is light on the right side switched on
+  allowed: ['NO_EFFECT', 'GRADIENT_DRIVER', 'GRADIENT_PASSENGER']
```

Comment:
I would like to have a short comment on what is meant by gradient.

Does "GRADIENT_DRIVER" means that the effect is mostly (brightest) on driver side?
Does "NO_EFFECT" means "turned off", or rather "symmetrical effect"?

Reply:
Those effects' names are just given as examples for possible default values. The actual values in practice will be implementation- or OEM-specific.
- `GRADIENT_DRIVER` means that the light bars has more intensity on the driver's side, and the intensity fades down towards the passenger.
- `GRADIENT_PASSENGER` means that the light bars has more intensity on the passenger's side, and the intensity fades down towards the driver.
- `NO_EFFECT` means that the whole light bar is illuminated with the same intensity.

Reply:
If they are just examples and will vary then it maybe fits better as a comment.

Reply:
@jdacoello - if it is just examples I would like it to instead be a comment, similar to what was added in https://github.com/COVESA/vehicle_signal_specification/pull/535/files

Reply:
Example allowed values were removed. Information was added to the comment field.